Ported Events Schedule Poster component to Svelte5 (#50)

Kishan-Patel-dev opened the pull request:
- Migrated the Events Schedule Poster component to Svelte5.
- Replaced DB dependency with JSON file for data seeding.
- Maintained functionality as per the original component.
- Ready for review. @Sing-Li @Dnouv

Fixes #50

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/GSoC-Community-Hub?pullRequest=52) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/GSoC-Community-Hub?pullRequest=52) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/GSoC-Community-Hub?pullRequest=52) it.</sub>

anjy7 commented:
hey @Kishan-Patel-dev, thank you for your contribution. Please remove the yarn.lock file and share the screenshots of the UI in the PR. 

#47 PR adds the support of Domain-specific-language. Once that is merged, please use the same DSL to create a demo page of your component along with the seed data like rest of the components.

Kishan-Patel-dev commented:
Hey @anjy7 Can you please conform the following approach is Correct to Solve this issue Faster 
- Convert EventPoster Component to AMGL (All this components https://github.com/RocketChat/RC4Conferences/tree/main/app/components/conferences/eventPoster/components)
- Add the JSoN Seed Data (https://github.com/RocketChat/RC4Conferences/commit/51d1ad9c639fb635c17300572eee28f9bf24add3)
- Create a Demo Page

Sing-Li commented:
@Kishan-Patel-dev   I think you only need to:

- use the agml file to generate your existing .svelte component. (no need to port all the separate components)
- use the latest dataset for JSON; not the 2023 one   --- see https://gsoc.rocket.chat/community/conferences/c/190299#session

anjy7 commented:
@Kishan-Patel-dev You have already created the component mentioned in the issue. Next steps are - instead of modifying the page.svelte directly, create a .agml file that will create a page.svelte. You can refer main.agml and community.agml files that create the corresponding page.svelte files.

Kishan-Patel-dev commented:
Quick update: Almost done with integration, but facing a compilation issue
I have completed the component and just need to integrate it with the existing code. However, I'm currently blocked due to a compilation error while running bun run compile.
Once this is resolved, I can proceed with the final integration. Let me know how you'd like me to handle this. Thanks!

anjy7 commented:
> Quick update: Almost done left with integration due to compilation issue I have completed the component and just need to integrate it with the existing code. However, I'm currently blocked due to a compilation error while running bun run compile. Once this is resolved, I can proceed with the final integration. Let me know how you'd like me to handle this. Thanks!

could you please pull the latest changes and check again? If error still persists, paste it over here @Kishan-Patel-dev

Kishan-Patel-dev commented:
**All requested changes have been implemented!**
- Converted EventPoster component to AGML
- Added JSON seed data
- Integrated EventPoster into the top navigation menu

**Output**
![Screenshot From 2025-03-29 13-41-33](https://github.com/user-attachments/assets/b11b44a6-4245-4383-941f-1720d1cb27eb)
![Screenshot From 2025-03-29 13-41-50](https://github.com/user-attachments/assets/cf574cfa-1546-4097-8e27-729d5bf84704)
![Screenshot From 2025-03-29 13-42-03](https://github.com/user-attachments/assets/15180985-8c0d-4231-b17d-822f3f865f1e)

@Sing-Li & @anjy7 PR is ready for Review.

anjy7 commented:
@Kishan-Patel-dev please remove the package lock file..

Kishan-Patel-dev commented:
@Sing-Li @anjy7 
All requested changes have been implemented:
✅ Converted EventPoster component to AGML
✅ Added JSON seed data
✅ Integrated EventPoster into the top navigation menu
✅ Removed unnecessary files

All checks have passed, and there are no merge conflicts. Please review when you have time. Thanks! 🚀

anjy7 commented:
@Kishan-Patel-dev Thanks for the contributions and adjustments! I will review the code by tonight and merge it..

anjy7 commented:
@Kishan-Patel-dev merged, thank you for the contribution 🚀 🚀
